src/main/java/org/wildfly/security/auth/provider/ldap/OtpCredentialLoader.java

```diff
@@ -73,10 +80,14 @@ public ForIdentityLoader(DirContextFactory contextFactory, String distinguishedN
         }
 
         @Override
-        public SupportLevel getCredentialAcquireSupport(String credentialName) {
-            if ( ! OtpCredentialLoader.this.myCredentialName.equals(credentialName)) {
+        public SupportLevel getCredentialAcquireSupport(final Class<? extends Credential> credentialType, final String algorithmName) {
+            if (credentialType != PasswordCredential.class) {
                 return SupportLevel.UNSUPPORTED;
             }
+            if (algorithmName == null) {
```

Comment:
I don't think this 'if' is correct - we still need to verify that this identity does have the set of OTP attributes associated with it even though we don't care which algorithm is in use.